bindings/python/test/test_mongocrypt.py

```diff
@@ -223,6 +225,9 @@ def fetch_keys(self, filter):
     def insert_data_key(self, data_key):
         raise NotImplementedError
 
+    def bson_encode(self, doc):
+        return BSON.encode(doc)
```

Comment:
Yuck. You have bson.encode now and none of this will work with older PyMongo versions.

Reply:
I added the `bson_encode(self, doc) ` abstract method to deal with the fact that libmongocrypt requires the keyAltName to be passed as a BSON encoded document: https://jira.mongodb.org/browse/CDRIVER-3275. Instead of having pymongocrypt require pymongo I've implemented bson encoding via dependency injection.

This test implements a mock version of the callback API that pymongo will implement. In pymongo we can implement `bson_encode` using `bson.encode`.

> You have bson.encode now and none of this will work with older PyMongo versions.

I don't there's any concern for compatibility. This code uses the standard BSON.encode() API that has existed in pymongo forever. Only the tests for pymongocrypt require pymongo.